snapcraft/tests/test_sources.py

```diff
@@ -321,6 +321,58 @@ def test_init_with_source_branch_and_tag_raises_exception(self):
         self.assertEqual(raised.exception.message, expected_message)
 
 
+class TestSubversion(SourceTestCase):
+
+    def test_pull_remote(self):
+        svn = sources.Subversion('svn://my-source', 'source_dir')
+        svn.pull()
+        self.mock_run.assert_called_once_with(
+            ['svn', 'checkout', 'svn://my-source', 'source_dir'])
+
+    def test_pull_local(self):
+        svn = sources.Subversion(
+            'file://{}'.format(os.path.abspath('my-source/')), 'source_dir')
+        svn.pull()
+        self.mock_run.assert_called_once_with(
+            ['svn', 'checkout',
+             'file://{}'.format(os.path.abspath('my-source/')),
+             'source_dir'])
```

Comment:
notice that here you are not testing anything, because you are passing to the Subversion constructor a source that will return false in isdir.
On the constructor, pass only 'my-source'.